packages/main/src/plugin/kubernetes/kubernetes-port-forward-service.ts

```diff
@@ -0,0 +1,114 @@
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type { KubeConfig } from '@kubernetes/client-node';
+
+import type { KubernetesClient } from '/@/plugin/kubernetes/kubernetes-client.js';
+import { PortForwardConnectionService } from '/@/plugin/kubernetes/kubernetes-port-forward-connection.js';
+import type { ForwardConfig, UserForwardConfig } from '/@/plugin/kubernetes/kubernetes-port-forward-model.js';
+import { ConfigManagementService, MemoryBasedStorage } from '/@/plugin/kubernetes/kubernetes-port-forward-storage.js';
+import { ForwardConfigRequirements } from '/@/plugin/kubernetes/kubernetes-port-forward-validation.js';
+import type { IDisposable } from '/@/plugin/types/disposable.js';
+import { isFreePort } from '/@/plugin/util/port.js';
+
+/**
+ * Service provider for Kubernetes port forwarding.
+ * Entrypoint for the third party services to operate with port forwards.
+ * @see KubernetesPortForwardServiceProvider.getService
+ */
+export class KubernetesPortForwardServiceProvider {
+  /**
+   * Gets the port forward service for the given Kubernetes configuration.
+   * @param kubeClient the Kubernetes client
+   * @returns The port forward service.
+   */
+  getService(kubeClient: KubernetesClient): KubernetesPortForwardService {
+    const forwardingConnectionService = new PortForwardConnectionService(
+      kubeClient,
+      new ForwardConfigRequirements(isFreePort),
+    );
+    const forwardConfigStorage = new MemoryBasedStorage();
+    const configManagementService = new ConfigManagementService(forwardConfigStorage);
+    return new KubernetesPortForwardService(configManagementService, forwardingConnectionService);
+  }
+
+  /**
+   * Gets the configuration key for the given Kubernetes configuration.
+   * The key is used for caching services and also for storing information about created forwards linked to each service.
+   *
+   * @param kubeConfig - The Kubernetes configuration.
+   * @returns The configuration key.
+   */
+  protected getKubeConfigKey(kubeConfig: KubeConfig): string {
+    return kubeConfig.currentContext;
+  }
+}
+
+/**
+ * Service for managing Kubernetes port forwarding.
+ * @see KubernetesPortForwardServiceProvider.getService
+ */
+export class KubernetesPortForwardService {
```

Comment:
I would recommend to make `KubernetesPortForwardService` extend IDisposal, if we need to cleanup everything, the service should keep track of the disposals, and have `KubernetesPortForwardService#dispose` method to cleanup all port foward easily 

Currently the disposal is only provided through the `KubernetesPortForwardService#startForward` but nothing is saving them to cleanup resources.